lib/nodeFetchHttpClient.ts

```diff
@@ -46,7 +46,14 @@ export class NodeFetchHttpClient extends FetchHttpClient {
       httpRequest.headers.set("Cookie", cookieString);
     }
 
-    if (httpRequest.proxySettings) {
+    if (httpRequest.agentSettings) {
```

Comment:
It seems reasonable to me that (in the rase case) when `proxySettings` and `agentSettings` are both defined, the `agentSettings` should take precedence. I think this should be a very unlikely scenario, however I am open to adjusting precedence.

Reply:
I concur with you that `agentSettings` should take precedence.